api/views.py

```diff
@@ -289,6 +289,17 @@ def get(self, request):
         )
 
 
+class MailUpdateView(generics.UpdateAPIView):
+    def put(self, request, video_id):
+        video_object = Video.objects.get(id=video_id)
+        viewed_at = datetime.now()
+        video_object.viewed_at = viewed_at
+        video_object.save()
+        return response.Response(
+            data={"viewed_at": str(viewed_at)}, status=status.HTTP_200_OK
+        )
+
+
```

Comment:
Why not combine with `MailListView` to make `MailUpdateListView`

Reply:
If the client instead sends the `viewed_at` date, then it would be fairly straight forward to combine the two views.
You'd only need to add the `ListModelMixin` and override the `get_object` method to return the video with the specified ID.

I don't think this is worth the effort though. The endpoints need to be heavily refactored imo.

Reply:
Fair enough, and I agree